src/FreeDSx/Ldap/Control/Ad/DirSyncRequestControl.php

```diff
@@ -143,7 +143,11 @@ public function setCookie(string $cookie)
     }
 
     /**
-     * {@inheritdoc}
+     * @param AbstractType $type
+     * @return Control
+     * @throws ProtocolException
+     * @throws \FreeDSx\Asn1\Exception\EncoderException
+     * @throws \FreeDSx\Asn1\Exception\PartialPduException
```

Comment:
Was this an explicit change, or just how Psalm decided to do it? I think updating the interface this is coming from -- `ProtocolElementInterface` -- might be better. As in just adding additional throws to it. Helps avoid some duplicated docblock anyway. Though I can see why it potentially did this if it wants to always return the specific type in the docblock and not the abstract type from the interface

Reply:
It is just from Psalm telling that these Exceptions can be seen there.

This should be pre-existing confusing behaviour. Just previously undocumented.

Reply:
Ah, these \@inheritdoc replacements. Psalm is barely aware of the interaction between Interfaces/Traits and classes.

But sure. I can review them, though it is kind of tedious to review, and still correct.